modules/books_data_manipulation.rb

```diff
@@ -0,0 +1,34 @@
+require 'json'
+
+module BooksDataManipulation
+  def save_books_to_json(file_path, new_book)
+    parsed_data = []
+
+    if File.exist?(file_path)
+      current_data = File.read(file_path)
+      parsed_data = JSON.parse(current_data) unless current_data.empty?
+    end
+
+    book_data = {
+      title: new_book.title,
+      author: new_book.author
+    }
+
+    parsed_data << book_data
+
+    File.write(file_path, JSON.pretty_generate(parsed_data))
+  end
+
+  def load_books_from_json(file_path)
+    if File.exist?(file_path)
+      current_data = File.read(file_path)
+      parsed_data = JSON.parse(current_data) unless current_data.empty?
+      return [] if parsed_data.nil?
+    end
+
+    parsed_data.map do |book_data|
+      book = Book.new(book_data['title'], book_data['author'])
+      book
+    end
+  end
```

Comment:
- [ ] Hey there! 👋 Great job on implementing the book-saving method. 📚 I noticed you're checking if the books.json file exists before reading from it, which is awesome. However, one thing to consider for better robustness: it might be a good idea to refactor the code to create the db directory and the books.json file if they don't exist.

This adjustment can prevent potential errors and ensure a smoother user experience. 🚀 Keep up the good work! If you have any questions or need further clarification, feel free to ask. 👍

![Screenshot from 2023-11-22 18-13-48](https://github.com/IvonneBenitesRodriguez/oop-school-library/assets/112261237/72b4a6c7-ba53-4fb3-a996-798415fd7a77)

Reply:
Dear @zeff96,
Very nice to meet you, 
Thanks for the suggestions, 
can you please give us a moment in zoom to clarify what you are requesting us.
Since, we think our app meet the projects requirements.
We wrote you in slack and in zoom also.  
We wait for your kind reply, 
Thanks in advance